sdk/python/kfp/compiler/_op_to_template.py

```diff
@@ -253,8 +253,12 @@ def _op_to_template(op: BaseOp):
         if processed_op.pod_labels:
             template['metadata']['labels'] = processed_op.pod_labels
     # retries
-    if processed_op.num_retries:
-        template['retryStrategy'] = {'limit': processed_op.num_retries}
+    if processed_op.num_retries or processed_op.retry_policy:
```

Comment:
Curious what is the intended behavior if `retry_policy` is set but not `num_retries`? 

If in that case the `retryPolicy` is a no-op then should we consider warning/raising error earlier so that users can correct it? Also I think that'll slightly simplify the code structure here

Reply:
I've added a warning message. 
I don't see how it can simplify the code structure. Can you elaborate?

Reply:
What I meant was, if we add the early warning/error in `_container_op.py` in `set_retry_policy` and `set_retry`, then we can do the following here:

```python
if processed_op.num_retries:
  template['retryStrategy'] = {}
  template['retryStrategy']['limit'] = processed_op.num_retries
if processed_op.retry_policy:
  template['retryStrategy']['retryPolicy'] = processed_op.retry_policy
```

Reply:
Hm, I don't think an early warning in `set_retry_policy` is a good idea. Consider the following fully legit code.
```python 
op = ContainerOp(...).set_retry_policy('Always').set_retry(5)
```
In this case, `set_retry_policy` could've emitted a warning message confusing the user.

Reply:
Ah I see.

WDYT if we make `retry_policy` a parameter of `set_retry`, which defaults to `None`?

Reply:
You mean this?
```python
def set_retry(self, num_retries: int, policy: str):
    self.num_retries = num_retries
    self.retry_policy = policy
```
Actually, this works and makes the interface simpler. 

However, simplifying the compiler code does not look safe to me. The user still can do something not very clever, like:
```python
op = ContainerOp(...).set_retry(5, 'Always')
op.num_retries = None
# or
op2 = ContainerOp(...).set_retry(None, 'Always')
```

Reply:
I've pushed the simplified interface.

Reply:
Thanks. Then it LGTM